daemon/list.go

```diff
@@ -90,6 +90,41 @@ func (daemon *Daemon) Containers(config *types.ContainerListOptions) ([]*types.C
 	return daemon.reduceContainers(config, daemon.transformContainer)
 }
 
+func (daemon *Daemon) filterByNameIDMatches(ctx *listContext) []*container.Container {
+	names := ctx.filters.Get("name")
+	ids := ctx.filters.Get("id")
+	if len(names)+len(ids) == 0 {
+		// if name or ID filters are not in use, return to
+		// standard behavior of walking the entire container
+		// list from the daemon's in-memory store
+		return daemon.List()
+	}
+
+	matches := make(map[string]bool)
+	// look for name matches; errors only report not found and can be ignored
+	if len(names) > 0 {
+		for id, idNames := range ctx.names {
+			for _, eachName := range idNames {
+				if ctx.filters.Match("name", eachName) {
+					matches[id] = true
+				}
+			}
+		}
+	}
+	// look for ID matches; errors are not found and can be ignored
```

Comment:
Hum not really, if we reach this part of the code, we are guaranteed that `len(ids) > 0` is always `true` :angel:.